File: eval_front/src/app/data/models/detalle.ts
import { Producto } from "./producto"
import { Transaccion } from "./transaccion"

export class Detalle{
    constructor(
        public detalle_id:string,
        public detalle_cantidad:number,
        public transaccion:Transaccion,
        public producto:Producto
    ){}
}

File: eval_back/ProductosService/App/Models/Detalle.cs
using System.ComponentModel.DataAnnotations;

namespace ProductosService.App.Models
{
    public class Detalle
    {
        [Key]
        public Guid detalle_id { get; set; }
        public int detalle_cantidad { get; set; }

        public Guid tran_id { get; set; }
        public Guid prod_id { get; set; }

        public Transaccion transaccion { get; set; }
        public Producto producto { get; set; }
    }
}
